core/src/main/java/io/undertow/util/LegacyCookieSupport.java

```diff
@@ -0,0 +1,389 @@
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+package io.undertow.util;
+
+import io.undertow.UndertowMessages;
+import io.undertow.server.handlers.Cookie;
+
+import java.util.Date;
+
+/**
+ * Class that contains static constants and utility methods for legacy Set-Cookie format.
+ * Porting from JBossWeb and Tomcat code.
+ */
+public final class LegacyCookieSupport {
+
+    // --------------------------------------------------------------- Constants
+
+    /**
+     * If set to true, we parse cookies strictly according to the servlet,
+     * cookie and HTTP specs by default.
+     */
+    private static final boolean STRICT_SERVLET_COMPLIANCE;
+
+    /**
+     * If true, separators that are not explicitly dis-allowed by the v0 cookie
+     * spec but are disallowed by the HTTP spec will be allowed in v0 cookie
+     * names and values. These characters are: \"()/:<=>?@[\\]{} Note that the
+     * inclusion of / depend on the value of {@link #FWD_SLASH_IS_SEPARATOR}.
+     */
+    private static final boolean ALLOW_HTTP_SEPARATORS_IN_V0;
+
+    /**
+     * If set to false, we don't use the IE6/7 Max-Age/Expires work around.
+     * Default is usually true. If STRICT_SERVLET_COMPLIANCE==true then default
+     * is false. Explicitly setting always takes priority.
+     */
+    private static final boolean ALWAYS_ADD_EXPIRES;
+
+    /**
+     * If set to true, the <code>/</code> character will be treated as a
+     * separator. Default is false.
+     */
+    private static final boolean FWD_SLASH_IS_SEPARATOR;
+
+    /**
+     * The list of separators that apply to version 0 cookies. To quote the
+     * spec, these are comma, semi-colon and white-space. The HTTP spec
+     * definition of linear white space is [CRLF] 1*( SP | HT )
+     */
+    private static final char[] V0_SEPARATORS = {',', ';', ' ', '\t'};
+    private static final boolean[] V0_SEPARATOR_FLAGS = new boolean[128];
+
+    /**
+     * The list of separators that apply to version 1 cookies. This may or may
+     * not include '/' depending on the setting of
+     * {@link #FWD_SLASH_IS_SEPARATOR}.
+     */
+    private static final char[] HTTP_SEPARATORS;
+    private static final boolean[] HTTP_SEPARATOR_FLAGS = new boolean[128];
+
+    private static final String ancientDate;
+
+    static {
+
+        STRICT_SERVLET_COMPLIANCE = Boolean.valueOf(System.getProperty(
```

Comment:
Do we actually need all these flags? And if so should they be an Option rather than a system property?
(Undertow does not really go in for config via system properties) 

STRICT_SERVLET_COMPLIANCE in particular looks out of place as this is in the core rather than the servlet module.